docs/fixed_agg_workflow.md

```diff
@@ -0,0 +1,196 @@
+Thus, the first four nodes are the root (in cladecombiner parlance, an empty string, `""`), B, B.1, B.1.1, and B.1.1.529 (aka BA, but note that cladecombiner is pedantic about names and does not accept naked aliases like BA).
+Of these, only BA has multiple children among the observed lineages, so only it is a branching point in the tree.
+
+## Aggregation targets
+
+We also need the desired aggregation targets, the taxa to which we want to aggregate.
+As with observed taxa, we can read these in or create them from strings.
+Let us assume we want to aggregate to BA.2.86.1 (aka JN), BA.2, BA.5, and KP.1.1.
+
+Aggregation requires us to be explicit about whether we are truly aggregating or not.
+Three of these represent actual aggregations of the observed taxa, as we have observed lineages which are children of BA.2, BA.5, and JN.
+However, as we have no children of KP.1.1 in our observed lineages, it is a tip.
+
+```
+target_taxa = [
+    cladecombiner.Taxon("BA.2", False),
+    cladecombiner.Taxon("BA.2.86.1", False),
+    cladecombiner.Taxon("BA.5", False),
+    cladecombiner.Taxon("KP.1.1", True),
+]
+```
+
+## Aggregate
+
+We are now ready to create an aggregation object and aggregate.
+
+```
+agg = cladecombiner.BasicPhylogeneticAggregator(targets=target_taxa, taxonomy_scheme=scheme)
+
+res = agg.aggregate(input_taxa)
+```
+
+The `res` object is an `Aggregation` object, essentially just a `dict[Taxon, Taxon]`, which maps each of the observed lineages in `taxa` to some aggregated taxon.
+For the above input taxa and targets, the resulting mapping is:
+
+```
+Taxon(KP.1.1, tip=True)    : Taxon(KP.1.1, tip=True)
+Taxon(BA.5.2.6, tip=True)  : Taxon(BA.5, tip=False)
+Taxon(BQ.1.1.23, tip=True) : Taxon(BA.5, tip=False)
+Taxon(JN.1.9.1, tip=True)  : Taxon(BA.2.86.1, tip=False)
+Taxon(XDQ, tip=True)       : Taxon(BA.2.86.1, tip=False)
+Taxon(JN.9, tip=True)      : Taxon(BA.2.86.1, tip=False)
+Taxon(JN.6, tip=True)      : Taxon(BA.2.86.1, tip=False)
+Taxon(XCU, tip=True)       : Taxon(BA.2, tip=False)
+Taxon(BA.2.86, tip=True)   : Taxon(BA.2, tip=False)
+Taxon(FW.1.1.1, tip=True)  : Taxon(BA.2, tip=False)
+Taxon(BA.2, tip=True)      : Taxon(BA.2, tip=False)
+```
+
+## Using and saving the aggregation
+
+The resulting mapping can be used to re-name taxa in, for example, line-list data.
+A `dict[str,str]` can be obtained via `res.to_str()` which can be fed into the `replace` methods in [pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.replace.html) or [polars](https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.replace.html).
+
+The `res.to_str()` dictionary can be saved for posterity by writing to file using python's [`json.dumps()`](https://docs.python.org/3/library/json.html) function.
+
+Saving the python script for posterity is also recommended.
```

Comment:
On the call today we talked a little bit about pickling. The trick with pickling anything other than basic Python objects is that you can run into versioning trouble, IIRC. A problem for another time.